WebDriverAgentLib/Utilities/FBImageIOScaler.m

```diff
@@ -140,7 +140,9 @@ - (nullable NSData *)scaledImageWithImage:(NSData *)image
   uiImage = [UIImage imageWithCGImage:(CGImageRef)uiImage.CGImage
                                 scale:uiImage.scale
                           orientation:orientation];
-  [uiImage drawInRect:CGRectMake(0, 0, scaledSize.width, scaledSize.height)];
+  if (fabs(1.0 - scalingFactor) > FLT_EPSILON) {
```

Comment:
> Hm... UIImagePNGRepresentation(resultImage) generates (width = 828, height = 1792) image even the given resultImage was (width = 1792, height = 828) in landscape without UIGraphicsBeginImageContext, UIGraphicsGetImageFromCurrentImageContext and UIGraphicsEndImageContext...
> So, it seems we should keep them (can skip only drawInRect) to reflect orientation.

If this is the case then lets keep it as it was. This drawing operation is cheap.
Thanks for checking it

Reply:
```objc
  UIImage *uiImage = [UIImage imageWithData:image];
  CGSize size = uiImage.size;

  UIImageOrientation orientation = uiImage.imageOrientation;
#if !TARGET_OS_TV
  if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationPortrait) {
    orientation = UIImageOrientationUp;
  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationPortraitUpsideDown) {
    orientation = UIImageOrientationDown;
  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationLandscapeLeft) {
    orientation = UIImageOrientationRight;
  } else if (FBConfiguration.screenshotOrientation == UIInterfaceOrientationLandscapeRight) {
    orientation = UIImageOrientationLeft;
  }
#endif
  uiImage = [UIImage imageWithCGImage:(CGImageRef)uiImage.CGImage
                                scale:uiImage.scale
                          orientation:orientation];
  UIImage *resultImage;
  if (fabs(1.0 - scalingFactor) > FLT_EPSILON) {
    CGSize scaledSize = CGSizeMake(size.width * scalingFactor, size.height * scalingFactor);
    UIGraphicsBeginImageContext(scaledSize);
    [uiImage drawInRect:CGRectMake(0, 0, scaledSize.width, scaledSize.height)];
    resultImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
  } else {
    resultImage = uiImage;
  }
```

Reply:
Did you try something like that ^ ? If not please confirm it does not work as expected.

Reply:
Yea. In the case also before `UIImagePNGRepresentation` was expected, but `UIImagePNGRepresentation(resultImage)` generated portrait image... 🤯

Reply:
I observed the below case as well...
```
UIImage *uiImage = [UIImage imageWithData:image];
uiImage // (width = 828, height = 1792) 
UIImagePNGRepresentation(uiImage) // (width = 1792, height = 828)
```

debugger
```
(lldb) po [UIImage imageWithData:UIImagePNGRepresentation(uiImage)]
<UIImage:0x6000039e5200 anonymous {828, 1792}>

(lldb) po uiImage
<UIImage:0x6000039ed440 anonymous {1792, 828}>
```